core/store/src/trie/mod.rs

```diff
@@ -683,13 +695,17 @@ impl Trie {
 
     pub(crate) fn convert_to_insertions_and_deletions(
         changes: HashMap<CryptoHash, (Vec<u8>, i32)>,
-    ) -> (Vec<(CryptoHash, Vec<u8>, u32)>, Vec<(CryptoHash, Vec<u8>, u32)>) {
+    ) -> (Vec<TrieRefcountChange>, Vec<TrieRefcountChange>) {
         let mut deletions = Vec::new();
         let mut insertions = Vec::new();
         for (key, (value, rc)) in changes.into_iter() {
             match rc.cmp(&0) {
-                Ordering::Greater => insertions.push((key, value, rc as u32)),
-                Ordering::Less => deletions.push((key, value, (-rc) as u32)),
+                Ordering::Greater => {
+                    insertions.push(TrieRefcountChange { key_hash: key, value, rc: rc as u32 })
+                }
+                Ordering::Less => {
+                    deletions.push(TrieRefcountChange { key_hash: key, value, rc: (-rc) as u32 })
```

Comment:
This feels a bit weird. Why do we want to cast it as u32?

Reply:
Here changes are separated to insertions and deletions.
Not sure why it is needed, but for insertion or deletion `refcount` change number is always positive. `u32` seems suitable to emphasize this fact.

Reply:
> refcount change number is always positive

Not entirely sure what you mean. Do you mean that the result is always nonnegative? That does not change the fact that deletion will decrease rc, which means that it is negative here. Also later on we cast it back to i32 to calculate the resulting rc. So I am not sure why we need this casting here.

Reply:
Discussed offline